Disable Android e2e tests while we investigate foundational breakage

dcalhoun opened the pull request:
## Description
A `adb: no devices/emulators found` error is occurring and cause every run to fail. We intend to temporarily disable this job while we investigate the cause to avoid causing disruptions or confusion for contributors.

## How has this been tested?
Verify all CI checks succeed.

## Screenshots <!-- if applicable -->
n/a

## Types of changes
Tooling change.

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. <!-- Check code: `npm run lint`, Guidelines: https://developer.wordpress.org/coding-standards/wordpress-coding-standards/javascript/ -->
- [x] My code follows the accessibility standards. <!-- Guidelines: https://developer.wordpress.org/coding-standards/wordpress-coding-standards/accessibility/ -->
- [ ] I've tested my changes with keyboard and screen readers. <!-- Instructions: https://github.com/WordPress/gutenberg/blob/HEAD/docs/contributors/accessibility-testing.md -->
- [x] My code has proper inline documentation. <!-- Guidelines: https://developer.wordpress.org/coding-standards/inline-documentation-standards/javascript/ -->
- [x] I've included developer documentation if appropriate. <!-- Handbook: https://developer.wordpress.org/block-editor/ -->
- [x] I've updated all React Native files affected by any refactorings/renamings in this PR (please manually search all `*.native.js` files for terms that need renaming or removal). <!-- React Native mobile Gutenberg guidelines: https://github.com/WordPress/gutenberg/blob/HEAD/docs/contributors/code/native-mobile.md -->

dcalhoun committed:
Disable Android e2e tests while we investigate foundational breakage

A `adb: no devices/emulators found` error is occurring and cause every
run to fail. We intend to temporarily disable this job while we
investigate the cause to avoid causing disruptions or confusion for
contributors.

fluiddot commented:
I see in the PR checks that the Android e2e tests job is still failing, shouldn't have disappeared with the changes?

dcalhoun commented:
> I see in the PR checks that the Android e2e tests job is still failing, shouldn't have disappeared with the changes?

Yes, that was my intention. However, it looks like an empty `branches: []` configuration includes every branch, including this one, as indicated in the latest CI task run. I'll see if I can identify a better fix. 

<img width="859" alt="Screen Shot 2021-06-23 at 12 33 37" src="https://user-images.githubusercontent.com/438664/123142736-5d572900-d41f-11eb-8c5b-f739261e8071.png">

fluiddot commented:
> > I see in the PR checks that the Android e2e tests job is still failing, shouldn't have disappeared with the changes?
> 
> Yes, that was my intention. However, it looks like an empty `branches: []` configuration includes every branch, including this one, as indicated in the latest CI task run. I'll see if I can identify a better fix.
> 
> <img alt="Screen Shot 2021-06-23 at 12 33 37" width="859" src="https://user-images.githubusercontent.com/438664/123142736-5d572900-d41f-11eb-8c5b-f739261e8071.png">

I think you could disable it by adding this to the workflow ([solution reference](https://github.community/t/how-can-i-disable-a-github-action/17049/6)):
`if: false # always skip job`

dcalhoun committed:
Revert "Disable Android e2e tests while we investigate foundational breakage"

This reverts commit edb581e64ced3f9af26512d04a8681598dcabf04.

dcalhoun committed:
Leverage if configuration to disable test

Previous approach did not work. Using the if configuration seems more
appropriate.

dcalhoun committed:
Fix conditional logic for GitHub Action

mchowning reviewed (approved):
Changes look good., so if CI is happy, then I am happy. 😄
